ledger-tool/src/bigtable.rs

```diff
@@ -336,6 +338,250 @@ pub async fn transaction_history(
+            {
+                Ok(bigtable) => bigtable,
+                Err(err) => {
+                    error!("Failed to connect to source bigtable: {:?}", err);
+                    return Err(Box::new(err));
+                }
+            }
+        }
+    };
+
+    let destination_bigtable = {
+        if args.destination_is_emulator {
+            match solana_storage_bigtable::LedgerStorage::new_for_emulator(
+                &args.destination_instance_name,
+                &args.destination_app_profile_id,
+                &args.destination_endpoint.unwrap(),
+                None,
+            ) {
+                Ok(bigtable) => bigtable,
+                Err(err) => {
+                    error!("Failed to connect to destination bigtable: {:?}", err);
+                    return Err(Box::new(err));
+                }
+            }
+        } else {
+            match solana_storage_bigtable::LedgerStorage::new_with_config(
+                solana_storage_bigtable::LedgerStorageConfig {
+                    read_only: true,
+                    timeout: None,
+                    credential_type: CredentialType::Filepath(Some(
+                        args.destination_credential_path.unwrap(),
+                    )),
+                    instance_name: args.destination_instance_name,
+                    app_profile_id: args.destination_app_profile_id,
+                },
+            )
+            .await
+            {
+                Ok(bigtable) => bigtable,
+                Err(err) => {
+                    error!("Failed to connect to destination bigtable: {:?}", err);
+                    return Err(Box::new(err));
+                }
+            }
+        }
+    };
+
+    let (s, r) = unbounded::<u64>();
+    for i in from_slot..to_slot {
+        s.send(i).unwrap();
+    }
+
+    let workers = min(to_slot - from_slot, num_cpus::get().try_into().unwrap());
+    debug!("worker num: {}", workers);
+
+    let success_slots = Arc::new(Mutex::new(vec![]));
+    let skip_slots = Arc::new(Mutex::new(vec![]));
+    let block_not_found_slots = Arc::new(Mutex::new(vec![]));
+    let failed_slots = Arc::new(Mutex::new(vec![]));
+
+    let tasks = (0..workers)
+        .map(|i| {
+            let r = r.clone();
+            let source_bigtable_clone = source_bigtable.clone();
+            let destination_bigtable_clone = destination_bigtable.clone();
+
+            let success_slots_clone = Arc::clone(&success_slots);
+            let skip_slots_clone = Arc::clone(&skip_slots);
+            let block_not_found_slots_clone = Arc::clone(&block_not_found_slots);
+            let failed_slots_clone = Arc::clone(&failed_slots);
+            tokio::spawn(async move {
+                while let Ok(slot) = r.try_recv() {
+                    debug!("worker {}: received slot {}", i, slot);
+
+                    if !args.force {
+                        match destination_bigtable_clone.get_confirmed_block(slot).await {
```

Comment:
we should really add a lighter "existence check" method to the API for cases like this where we don't actually need the block data.

Reply:
yeah... It seems more efficient.

Reply:
use another to implement it https://github.com/solana-labs/solana/issues/28166